flax/nnx/nnx/bridge/wrappers.py

```diff
@@ -143,16 +144,17 @@ def __call__(
       if 'params' not in _rngs and 'default' in _rngs:
         _rngs['params'] = _rngs.pop('default')
       out, variables = self.module.init_with_output(_rngs, *args, method=method, **kwargs)
-      def nn_var_to_nnx_state(kp, v):
-        assert isinstance(kp[0], jtu.DictKey)
-        vtype = variableslib.variable_type(kp[0].key)
-        return vtype(v)
-      for col, tree in jtu.tree_map_with_path(nn_var_to_nnx_state, variables).items():
+
+      nnx_vars = jtu.tree_map_with_path(bv.to_nnx_var, variables,
+                                        is_leaf=lambda x: isinstance(x, meta.AxisMetadata))
+      for col, tree in nnx_vars.items():
         self._setattr(col, tree)
         self.linen_collections.add(col)
```

Comment:
For `linen_collections` we should either implement this with a `list` or create a `PytreeSet`. The issue is that sets are not pytrees so they are treated as static but they are not hashable.

Reply:
`list` is mutable so not hashable either. Curious why that matters though... but I'll make it a tuple.